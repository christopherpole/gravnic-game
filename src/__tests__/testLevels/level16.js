import { MOVE_LEFT, MOVE_RIGHT, ENTITIES, levelIsComplete, makeMoves } from '../../index';

describe('Level 16', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.RAINBOW_BLOCK, id: 1 },
        },
        {
          staticEntity: { id: 4, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.GLASS, id: 3 },
        },
        { staticEntity: { id: 5, entityId: ENTITIES.BLACK_HOLE }, movableEntity: null },
        { staticEntity: { id: 6, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 8, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#FFFF00', id: 7 },
        },
      ],
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 10, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 9 },
        },
        { staticEntity: { id: 11, entityId: ENTITIES.LAVA }, movableEntity: null },
        { staticEntity: { id: 12, entityId: ENTITIES.STICKY_SPOT }, movableEntity: null },
        {
          staticEntity: { id: 14, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.SMART_BOMB, id: 13 },
        },
        { staticEntity: null, movableEntity: null },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_LEFT, MOVE_RIGHT]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
