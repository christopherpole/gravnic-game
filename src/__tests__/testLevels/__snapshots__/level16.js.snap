// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 16 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 13,
            "stuck": true,
          },
          "staticEntity": Object {
            "entityId": "STICKY_SPOT",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 3,
            "shrinking": true,
          },
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 5,
            "shrinking": true,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 9,
            "shrinking": true,
          },
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 11,
            "shrinking": false,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 13,
            "stuck": true,
          },
          "staticEntity": Object {
            "entityId": "STICKY_SPOT",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 11,
            "shrinking": false,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 13,
            "stuck": true,
          },
          "staticEntity": Object {
            "entityId": "STICKY_SPOT",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 11,
            "shrinking": false,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 13,
            "stuck": true,
          },
          "staticEntity": Object {
            "entityId": "STICKY_SPOT",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 11,
            "shrinking": false,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "id": 13,
            "stuck": true,
          },
          "staticEntity": Object {
            "entityId": "STICKY_SPOT",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "fading": true,
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "fading": true,
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 11,
            "shrinking": false,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "SMART_BOMB",
            "fading": true,
            "id": 13,
            "stuck": true,
          },
          "staticEntity": Object {
            "entityId": "STICKY_SPOT",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 11,
            "shrinking": false,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "STICKY_SPOT",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
  ],
]
`;
